### PR TITLE
DUOS-1314[risk=no]UI: Replace ajax methods that use deprecated APIs

### PR DESCRIPTION
**Pull request by raejohanek**
SCOPE:
- Remove unused methods from ajax
- Remove and replace usages of ajax methods that call deprecated APIs 
- - replace  getPartialDarRequestList with getDraftDarRequestList on researcher console
- -  replace getDarFields with describeDar on DataOwnerReview
- - replace registerStatus with Update on researcher review

UPDATE: Justin's comments are addressed in DUOS-1329. That PR must be merged first in order for the functionality on the researcherReview page to remain intact.

ADDRESSES:
https://broadworkbench.atlassian.net/browse/DUOS-1314

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/libs/ajax.js [hunk 1]
```diff
@@ -175,13 +175,6 @@ export const DAC = {
 
 export const DAR = {
 
-  describeDarWithConsent: async (darId) => {
```
Comment (raejohanek) [inline src/libs/ajax.js:178]: unused

File: src/libs/ajax.js [hunk 2]
```diff
@@ -254,12 +247,6 @@ export const DAR = {
     return darInfo;
   },
 
-  translateDataUse: async dataUse => {
-    const url = `${await Config.getOntologyApiUrl()}translate/summary`;
```
Comment (raejohanek) [inline src/libs/ajax.js:258]: unused

File: src/libs/ajax.js [hunk 3]
```diff
@@ -517,11 +496,6 @@ export const Election = {
     return res.json();
   },
 
-  findElectionById: async (electionId) => {
-    const url = `${await Config.getApiUrl()}/api/election/${electionId}`;
```
Comment (raejohanek) [inline src/libs/ajax.js:521]: unused

File: src/libs/ajax.js [hunk 4]
```diff
@@ -616,18 +590,6 @@ export const Election = {
     return await res.json();
   },
 
-  findElectionReviewById: async (electionId, referenceId) => {
-    if (electionId !== undefined) {
```
Comment (raejohanek) [inline src/libs/ajax.js:620]: unused

File: src/libs/ajax.js [hunk 5]
```diff
@@ -989,12 +945,6 @@ export const Votes = {
     return res.json();
   },
 
-  getDarFinalAccessVote: async (requestId) => {
-    const url = `${await Config.getApiUrl()}/api/dataRequest/${requestId}/vote/final`;
```
Comment (raejohanek) [inline src/libs/ajax.js:993]: unused

File: src/pages/ResearcherConsole.js [hunk 6]
```diff
@@ -135,7 +135,7 @@ class ResearcherConsole extends Component {
       }
     );
 
-    DAR.getPartialDarRequestList().then(
+    DAR.getDraftDarRequestList().then(
```
Comment (JVThomas) [inline src/pages/ResearcherConsole.js:138]: New method is working well, the data is populating on the table and is redirecting as it should

File: src/pages/ResearcherReview.js [hunk 7]
```diff
@@ -85,8 +85,11 @@ class ResearcherReview extends Component {
     } else if (voteStatus === "false") {
       status = "rejected";
     }
-    let userStatus = { status: status, rationale: rationale, roleId: 5 };
-    User.registerStatus(userStatus, this.props.match.params.dacUserId).then(
+    let updatedUser = this.state.user;
+    updatedUser.status = status;
+    updatedUser.rationale = rationale;
```
Comment (JVThomas) [inline src/pages/ResearcherReview.js]: The rationale isn't updating on submission, but I can see the rationale and status on the request payload. I suspect the UI is doing it's part but the back-end isn't, so the lack of functionality is not an issue here.

**Review by JVThomas — APPROVED** [review]
PR looks good. There's a bug for me where rationale isn't updating on the ResearcherReview console, but the rationale and status are in the payload within request, so I suspect it's an issue of the back-end not catching it (in which case it's not an issue for this UI PR).

File: src/pages/ResearcherReview.js (hunk 7 above)
Comment (raejohanek) [reply src/pages/ResearcherReview.js]: yeah I noticed that too do you think I should file a ticket?

Comment (JVThomas) [reply src/pages/ResearcherReview.js]: Yeah, without it admins can't vote to approve or disapprove of users.

Comment (JVThomas) [reply src/pages/ResearcherReview.js]: It may be a good idea to hold off on merging this for now, current develop has the rationale and vote updating correctly (because it's using the old endpoint). Once the rationale fix is in on the back-end this PR should function correctly.

**Review by JVThomas — DISMISSED** [review]
Nothing with the PR itself, just noting that the new back-end endpoints don't have a functioning update that supports rationale and status updates for `ResearcherReview`. Once a fix for that has been made in consent this PR should work as expected.

**Comment by raejohanek** [comment]
> Nothing with the PR itself, just noting that the new back-end endpoints don't have a functioning update that supports rationale and status updates for `ResearcherReview`. Once a fix for that has been made in consent this PR should work as expected.

I just tested this with local consent on branch DUOS-1329 and the status and rationale are updated on the user now

**Comment by rushtong** [comment]
I'd prefer that we marked PRs as DRAFT until they are ready to be tested against in the dev env.

File: src/libs/ajax.js [hunk 8]
```diff
@@ -300,15 +288,6 @@ export const DAR = {
...
 
-  getDarFields: async (id, fields) => {
-    let url = `${await Config.getApiUrl()}/api/dar/find/${id}`;
-    if (fields !== null) {
-      url = url + `?fields=${fields}`;
-    }
-    const res = await fetchOk(url, Config.authOpts());
-    return await res.json();
-  },
-
```
Comment (rushtong) [inline src/libs/ajax.js:311]: This means that we can remove the endpoint here: https://github.com/DataBiosphere/consent/blob/develop/src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResource.java#L94

File: src/libs/ajax.js (hunk 3 above)
Comment (rushtong) [reply src/libs/ajax.js:521]: This is a super handy method to keep in Consent, so let's not deprecate that endpoint.

File: src/libs/ajax.js [hunk 9]
```diff
@@ -616,18 +590,6 @@ export const Election = {
...
-  findElectionReviewById: async (electionId, referenceId) => {
-    if (electionId !== undefined) {
-      const url = `${await Config.getApiUrl()}/api/electionReview/${electionId}`;
-      const res = await fetchOk(url, Config.authOpts());
-      return await res.json();
-    } else {
-      const url = `${await Config.getApiUrl()}/api/electionReview/last/${referenceId}`;
-      const res = await fetchOk(url, Config.authOpts());
-      return await res.json();
-    }
```
Comment (rushtong) [inline src/libs/ajax.js:628]: This means we can remove https://github.com/DataBiosphere/consent/blob/develop/src/main/java/org/broadinstitute/consent/http/resources/ElectionReviewResource.java#L101 as well.

File: src/libs/ajax.js (hunk 5 above)
Comment (rushtong) [reply src/libs/ajax.js:993]: This means we can remove https://github.com/DataBiosphere/consent/blob/develop/src/main/java/org/broadinstitute/consent/http/resources/DataRequestVoteResource.java#L173

**Review by rushtong — APPROVED** [review]
Looks good. Let's file a ticket to remove the deprecated consent endpoints as noted in comments.

File: src/libs/ajax.js (hunk 8 above)
Comment (raejohanek) [reply src/libs/ajax.js:311]: this is already being deleted in DUOS-1315 which is now unblocked bc of this approval

File: src/libs/ajax.js (hunk 9 above)
Comment (raejohanek) [reply src/libs/ajax.js:628]: I just added this deletion to the 1315 PR

Comment (raejohanek) [reply src/libs/ajax.js:628]: I also added the deletion of the one you commented on below getDarFinalAccessVote, to the 1315 PR as well
